src/apps/filters/h3.c

```diff
@@ -1189,6 +1189,1238 @@ SUBCOMMAND(uncompactCells,
+    return E_SUCCESS;
+}
+
+SUBCOMMAND(getRes0Cells, "Returns all of the resolution 0 cells") {
+    Arg *args[] = {&getRes0CellsArg, &helpArg};
+    PARSE_SUBCOMMAND(argc, argv, args);
+    H3Index *out = calloc(122, sizeof(H3Index));
+    H3Error err = H3_EXPORT(getRes0Cells)(out);
+    if (err != E_SUCCESS) {
+        free(out);
+        return err;
+    }
+    printf("[");
+    bool hasPrinted = false;
+    for (int i = 0; i < 122; i++) {
+        if (out[i] > 0) {
+            if (hasPrinted) {
+                printf(", ");
+            }
+            printf("%" PRIx64, out[i]);
+            hasPrinted = true;
+        }
+    }
+    printf("]\n");
+    free(out);
+    return E_SUCCESS;
+}
+
+SUBCOMMAND(getPentagons,
+           "Returns all of the pentagons at the specified resolution") {
+    int res = 0;
+    Arg resArg = {.names = {"-r", "--resolution"},
+                  .required = true,
+                  .scanFormat = "%d",
+                  .valueName = "res",
+                  .value = &res,
+                  .helpText = "Resolution, 0-15 inclusive."};
+    Arg *args[] = {&getPentagonsArg, &resArg, &helpArg};
+    PARSE_SUBCOMMAND(argc, argv, args);
+    H3Index *out = calloc(12, sizeof(H3Index));
+    H3Error err = H3_EXPORT(getPentagons)(res, out);
+    if (err != E_SUCCESS) {
+        free(out);
+        return err;
+    }
+    printf("[");
+    bool hasPrinted = false;
+    for (int i = 0; i < 12; i++) {
+        if (out[i] > 0) {
+            if (hasPrinted) {
+                printf(", ");
+            }
+            printf("%" PRIx64, out[i]);
+            hasPrinted = true;
+        }
+    }
+    printf("]\n");
+    free(out);
+    return E_SUCCESS;
+}
+
+SUBCOMMAND(pentagonCount, "Returns 12") {
+    Arg *args[] = {&pentagonCountArg, &helpArg};
+    PARSE_SUBCOMMAND(argc, argv, args);
+    printf("12\n");
+    return E_SUCCESS;
+}
+
+SUBCOMMAND(greatCircleDistanceRads,
+           "Calculates the 'great circle' or 'haversine' distance between two "
+           "lat, lng points, in radians") {
+    char filename[1024] = {0};  // More than Windows, lol
+    Arg filenameArg = {.names = {"-f", "--file"},
+                       .scanFormat = "%1023c",
+                       .valueName = "FILENAME",
+                       .value = &filename,
+                       .helpText =
+                           "The file to load the coordinates from. Use -- to "
+                           "read from stdin."};
+    char coordinateStr[1501] = {0};
+    Arg coordinateStrArg = {
+        .names = {"-c", "--coordinates"},
+        .scanFormat = "%1500c",
+        .valueName = "ARRAY",
+        .value = &coordinateStr,
+        .helpText =
+            "The array of coordinates to convert. Up to 1500 characters."};
+    Arg *args[] = {&greatCircleDistanceRadsArg, &filenameArg, &coordinateStrArg,
+                   &helpArg};
+    PARSE_SUBCOMMAND(argc, argv, args);
+    if (!filenameArg.found && !coordinateStrArg.found) {
```

Comment:
```suggestion
    if (filenameArg.found == coordinateStrArg.found) {
```

both shouldn't be provided, right?

Reply:
As mentioned in the prior PR, no. Either a file is provided or the coordinates are provided in-line in the call, not both.